tosfs/core.py

```diff
@@ -203,6 +203,59 @@ def info(
 
         return self._try_dir_info(bucket, key, path, fullpath)
 
+    def rmdir(self, path: str) -> None:
+        """Remove a directory if it is empty.
+
+        Parameters
+        ----------
+        path : str
+            The path of the directory to remove. The path should be in the format
+            `tos://bucket_name/directory_path`.
+
+        Raises
+        ------
+        FileNotFoundError
+            If the directory does not exist.
+        NotADirectoryError
+            If the path is not a directory.
+        TosfsError
+            If the directory is not empty,
+             or the path is a bucket.
+
+        Examples
+        --------
+        >>> fs = TosFileSystem()
+        >>> fs.rmdir("tos://mybucket/mydir/")
+
+        """
+        path = self._strip_protocol(path).rstrip("/") + "/"
+        bucket, key, _ = self._split_path(path)
+        if not key:
+            logger.error("Cannot remove a bucket using rmdir api.")
+            raise TosfsError("Cannot remove a bucket using rmdir api.")
+
+        if not self.exists(path):
+            raise FileNotFoundError(f"Directory {path} not found.")
+
+        if not self.isdir(path):
+            raise NotADirectoryError(f"{path} is not a directory.")
+
+        if len(self.ls(path, refresh=True, detail=False)) > 0:
+            raise TosfsError(f"Directory {path} is not empty.")
+
+        try:
+            self.tos_client.delete_object(bucket, key.rstrip("/") + "/")
+            self.invalidate_cache(path.rstrip("/"))
+        except tos.exceptions.TosClientError as e:
+            logger.error("Tosfs failed with client error: %s", e)
```

Comment:
Similar issue, if we decide to raise the error, then we are not required to log the message everywhere.

Reply:
make sense

Reply:
I only cleared unnecessary logs belonging to this PR. And filed an issue to track this work for other apis. #26